src/main/java/com/exactpro/th2/sim/impl/Simulator.java

```diff
@@ -177,8 +177,11 @@ public void touchRule(TouchRequest request, StreamObserver<Empty> responseObserv
             return;
         }
 
+        logger.debug("Call touch on rule with id = {}", ruleInfo.getId());
+
         try {
             ruleInfo.touch(ObjectUtils.defaultIfNull(request.getArgsMap(), Collections.emptyMap()));
+            responseObserver.onNext(Empty.newBuilder().build());
```

Comment:
Can't we use Empty.getDefaultInstance() ?
```suggestion
            responseObserver.onNext(Empty.getDefaultInstance());
```